src/test/java/de/tum/in/www1/artemis/service/CourseScoreCalculationServiceTest.java

```diff
@@ -127,42 +130,109 @@ void calculateCourseScoreForStudentWithMultipleResultsInParticipation(boolean wi
         assertThat(result.getScore()).isEqualTo(0.0);
         result.score(null);
 
-        var studentScoreResult = courseScoreCalculationService.calculateCourseScoreForStudent(student.getId(), studentParticipations, 25.0, 5.0,
-                new PlagiarismCaseService.PlagiarismMapping(Collections.emptyMap()));
-        assertThat(studentScoreResult.studentId()).isEqualTo(student.getId());
-        assertThat(studentScoreResult.relativeScore()).isEqualTo(16.0);
-        assertThat(studentScoreResult.absolutePoints()).isEqualTo(4.0);
-        assertThat(studentScoreResult.currentRelativeScore()).isEqualTo(80.0);
-        assertThat(studentScoreResult.achievedPresentationScore()).isEqualTo(0);
-        assertThat(studentScoreResult.presentationScorePassed()).isFalse();
-        assertThat(studentScoreResult.mostSeverePlagiarismVerdict()).isNull();
-        assertThat(studentScoreResult.getAbsolutePointsEligibleForBonus()).isEqualTo(0.0);
+        StudentScoresDTO studentScoresDTO = courseScoreCalculationService.calculateCourseScoreForStudent(course, student.getId(), studentParticipations,
+                new CourseScoreCalculationService.MaxAndReachablePoints(25.0, 5.0), List.of());
+        if (withDueDate) {
+            assertThat(studentScoresDTO.absoluteScore()).isEqualTo(2.1);
+            assertThat(studentScoresDTO.relativeScore()).isEqualTo(8.4);
+            assertThat(studentScoresDTO.currentRelativeScore()).isEqualTo(42.0);
+        }
+        else {
+            assertThat(studentScoresDTO.absoluteScore()).isEqualTo(4.6);
+            assertThat(studentScoresDTO.relativeScore()).isEqualTo(18.4);
+            assertThat(studentScoresDTO.currentRelativeScore()).isEqualTo(92.0);
+        }
+
+        Map<Long, BonusSourceResultDTO> bonusSourceResultDTOMap = courseScoreCalculationService.calculateCourseScoresForExamBonusSource(course.getId(), List.of(student.getId()));
+
+        assertThat(bonusSourceResultDTOMap).hasSize(1);
+        BonusSourceResultDTO bonusSourceResultDTO = bonusSourceResultDTOMap.get(student.getId());
+        assertThat(bonusSourceResultDTO.achievedPoints()).isEqualTo(0.0);
+        assertThat(bonusSourceResultDTO.achievedPresentationScore()).isEqualTo(0);
+        assertThat(bonusSourceResultDTO.mostSeverePlagiarismVerdict()).isNull();
+    }
+
+    @Test
+    @WithMockUser()
```

Comment:
Does this do anything? Without any mock user given?

Check all occurrences in case you remove it.